### PR TITLE
Alexey 99 websockets

### PR DESCRIPTION
**Pull request by rezniknn**

### CODE REVIEW

**Comment by mansuleman** [comment]
Need tests for WebSocketHandler

File: static/js/main.js [hunk 1]
```diff
@@ -204,7 +209,24 @@ var ep = window['ep'] || {
...
+        }
+    },
+
+    parseWSMessage: function (message) {
+        console.log("Message from ws: " + message);
+        var json = JSON.parse(message);
+        switch (json.msg_type) {
+            case 'MODULE_UPDATE':
+                //Request to update Module info has been received
+                ep._updateModuleUI(json.slot, json.module_name);
```
Comment (user2589) [inline static/js/main.js:225]: what if slot is bigger than number of available slots?

File: static/js/main.js [hunk 2]
```diff
@@ -204,7 +209,24 @@ var ep = window['ep'] || {
...
+        var lock_container = $("#module_header_"+slot_id).find(".module_lock");
+        if (username) {
+            lock_container.text(ep._username_alias);
+        } else {
+            //Module is not locked by anyone
+            lock_container.empty();
+        }
+    },
+
+    parseWSMessage: function (message) {
```
Comment (user2589) [inline static/js/main.js]: it is not public method, name should start with underscore

**Comment by user2589** [comment]
Code itself is really good, I don't see major issues with it.
Feature is not complete, though. Besides unittests mentioned by @mansuleman, it is missing websockets for data returned by module, lock/unlock events, and documentation (it was in original task).
I assume we can accept this code into trunk, but task still has to be completed

File: static/js/main.js (hunk 2 above)
Comment (rezniknn) [inline static/js/main.js]: None of the existing methods do a safety check on slot_id. If you think it's necessary in this case, we should do it in python code of handlers

**Pull request closed without merging**